edx_exams/apps/api/v1/views.py

```diff
@@ -485,3 +488,49 @@ def post(self, request):
 
         data = {'exam_attempt_id': exam_attempt_id}
         return Response(data)
+
+
+class CourseExamAttemptView(ExamsAPIView):
+    """
+    Endpoint for getting timed or proctored exam and its attempt data given the request user.
+    /exam/attempt/course_id/{course_id}/content_id/{content_id}
+    Supports:
+        HTTP GET:
+
+            Returns an existing exam (by course_id and content id) with a nested attempt object,
+            if any attempt for that exam exists.
+            {
+                'exam': {
+                    'attempt': {...}
+                    ...
+                },
+            }
+    """
+
+    def get(self, request, course_id, content_id):
+        """
+        HTTP GET handler. Returns exam and an attempt, if one exists for the exam
+        """
+        exam = get_exam_by_content_id(course_id, content_id)
+
+        if exam is None:
+            data = {'exam': {}}
+            return Response(data)
+
+        exam_type_class = get_exam_type(exam['exam_type'])
+
+        # the following are additional fields that the frontend expects
+        exam['type'] = exam['exam_type']
```

Comment:
I'm thinking it might be better to create a custom exam serializer as well, instead of adding these fields here. Any thoughts?

Reply:
eh could go either way on this. Since we're building up a specific response that isn't based on just one model and isn't going to be reused it might be easier to read if the full response is just constructed here